File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    arr = deque()
    for i in range(len(priorities)):
         arr.append((priorities[i], i))
    
    result = []
    while arr:
        x = arr.popleft()
        for i in arr:
            if i[0] > x[0]:
                arr.append((x[0], x[1]))
                break
        else:
            result.append(x[1])
            
    
    for i in range(len(result)):
        if result[i] == location:
            answer = i + 1

    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    
    let arr = []
    
    priorities.forEach((x, idx) => {
        arr.push([x, idx]) 
    })
    
    let result = []
    while (arr.length) {
        let x = arr.shift()
        let check = 0
        for (let i of arr) {
            if (x[0] < i[0]) {
                check = 1
            }
        }
        
        if (check === 1) {
            arr.push(x)
        } else {
            result.push(x[1])
        }    
    }

    result.forEach((i, idx) => {
        if (i === location) {
            answer = idx+1
            return
        }
    })
    
    return answer;
}